File: Python3/Find the Index of the First Occurrence in a String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        m = len(needle)
        n = len(haystack)

        for start in range(n - m + 1):
            for i in range(m):
                if needle[i] != haystack[start+i]:
                    break
                if i == m - 1:
                    return start
        
        return -1

File: Golang/Find the Index of the First Occurrence in a String.go
func strStr(haystack string, needle string) int {
    m := len(needle)
    n := len(haystack)

    for start:=0;start < n - m + 1;start++ {
        for i:=0;i < m;i++ {
            if needle[i] != haystack[start+i] {
                break
            }
            if i == m - 1 {
                return start
            }
        }
    }
    return -1
}